share/eds/eds.go

```diff
@@ -214,6 +216,43 @@ func rootsToCids(eds *rsmt2d.ExtendedDataSquare) ([]cid.Cid, error) {
 	return rootCids, nil
 }
 
+// ReadEDS reads the first EDS quadrant (1/4) from an io.Reader CAR file.
+// Only the first quadrant will be read, which represents the original data.
+// The returned EDS is guaranteed to be full and valid against the DataRoot, otherwise ReadEDS errors.
+func ReadEDS(ctx context.Context, r io.Reader, root share.Root) (*rsmt2d.ExtendedDataSquare, error) {
+	carReader, err := car.NewCarReader(r)
+	if err != nil {
+		return nil, fmt.Errorf("failure to read car file: %w", err)
+	}
+
+	odsWidth := len(root.RowsRoots) / 2
+	odsSquareSize := odsWidth * odsWidth
+	shares := make([][]byte, odsSquareSize)
+	// the first quadrant is stored directly after the header,
+	// so we can just read the first odsSquareSize blocks
+	for i := 0; i < odsSquareSize; i++ {
+		block, err := carReader.Next()
+		if err != nil {
+			return nil, fmt.Errorf("failure to read next car entry: %w", err)
+		}
+		// the stored first quadrant shares are wrapped with the namespace twice.
+		// we cut it off here, because it is added again while importing to the tree below
```

Comment:
this needs a TODO for @walldiss when he completes the redundant namespace removal

Reply:
done